2025-09-03 14:50:26,915 - emsawd.core.logging_config - INFO - Logging configured successfully.
2025-09-03 14:50:26,915 - emsawd.core.logging_config - INFO - Log file created at: logs/historic_weather.log
2025-09-03 14:50:26,915 - root - INFO - Application starting up.
2025-09-03 14:51:03,937 - emsawd.core.logging_config - INFO - Logging configured successfully.
2025-09-03 14:51:03,938 - emsawd.core.logging_config - INFO - Log file created at: logs/historic_weather.log
2025-09-03 14:51:03,938 - root - INFO - Application starting up.
2025-09-03 14:51:04,282 - root - INFO - Main window shown. Starting event loop.
2025-09-03 14:54:09,429 - emsawd.core.logging_config - INFO - Logging configured successfully.
2025-09-03 14:54:09,429 - emsawd.core.logging_config - INFO - Log file created at: logs/historic_weather.log
2025-09-03 14:54:09,429 - root - INFO - Application starting up.
2025-09-03 14:54:09,645 - root - INFO - Main window shown. Starting event loop.
2025-09-03 14:56:03,450 - emsawd.core.services - INFO - Fetching weather data for cape town, period: 2025-08-27 to 2025-09-03, years: 5
2025-09-03 14:56:03,450 - emsawd.repositories.geocoding_repository - INFO - Requesting coordinates for 'cape town' from https://geocoding-api.open-meteo.com/v1/search
2025-09-03 14:56:04,285 - emsawd.repositories.geocoding_repository - INFO - API response status: 200
2025-09-03 14:56:04,286 - emsawd.repositories.geocoding_repository - INFO - Successfully found coordinates for 'cape town': (-33.92584, 18.42322)
2025-09-03 14:56:04,287 - emsawd.core.services - INFO - Coordinates for cape town: lat=-33.92584, lon=18.42322
2025-09-03 14:56:04,287 - emsawd.core.services - INFO - Fetching data for year offset 0 (2025): 2025-08-27 to 2025-09-03
2025-09-03 14:56:04,287 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 14:56:05,135 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 14:56:05,136 - emsawd.repositories.weather_repository - INFO - Successfully parsed 8 records from API response.
2025-09-03 14:56:05,137 - emsawd.core.services - INFO - Fetched 8 records for 2025
2025-09-03 14:56:05,137 - emsawd.core.services - INFO - Fetching data for year offset 1 (2024): 2024-08-27 to 2024-09-03
2025-09-03 14:56:05,137 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 14:56:06,222 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 14:56:06,222 - emsawd.repositories.weather_repository - INFO - Successfully parsed 8 records from API response.
2025-09-03 14:56:06,223 - emsawd.core.services - INFO - Fetched 8 records for 2024
2025-09-03 14:56:06,223 - emsawd.core.services - INFO - Fetching data for year offset 2 (2023): 2023-08-28 to 2023-09-04
2025-09-03 14:56:06,224 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 14:56:07,010 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 14:56:07,011 - emsawd.repositories.weather_repository - INFO - Successfully parsed 8 records from API response.
2025-09-03 14:56:07,012 - emsawd.core.services - INFO - Fetched 8 records for 2023
2025-09-03 14:56:07,013 - emsawd.core.services - INFO - Fetching data for year offset 3 (2022): 2022-08-28 to 2022-09-04
2025-09-03 14:56:07,013 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 14:56:09,458 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 14:56:09,458 - emsawd.repositories.weather_repository - INFO - Successfully parsed 8 records from API response.
2025-09-03 14:56:09,460 - emsawd.core.services - INFO - Fetched 8 records for 2022
2025-09-03 14:56:09,460 - emsawd.core.services - INFO - Fetching data for year offset 4 (2021): 2021-08-28 to 2021-09-04
2025-09-03 14:56:09,460 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 14:56:10,217 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 14:56:10,218 - emsawd.repositories.weather_repository - INFO - Successfully parsed 8 records from API response.
2025-09-03 14:56:10,220 - emsawd.core.services - INFO - Fetched 8 records for 2021
2025-09-03 14:56:10,221 - emsawd.core.services - INFO - Fetching data for year offset 5 (2020): 2020-08-28 to 2020-09-04
2025-09-03 14:56:10,221 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 14:56:10,966 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 14:56:10,967 - emsawd.repositories.weather_repository - INFO - Successfully parsed 8 records from API response.
2025-09-03 14:56:10,970 - emsawd.core.services - INFO - Fetched 8 records for 2020
2025-09-03 14:56:10,970 - emsawd.core.services - INFO - Total records collected: 48
2025-09-03 14:56:11,007 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,008 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,008 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,009 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,009 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,010 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,010 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,011 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,011 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,012 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,012 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,013 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,089 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:11,090 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:30,965 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:30,966 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:31,006 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:31,007 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,059 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,059 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,060 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,060 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,061 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,061 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,062 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,062 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,063 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,064 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,064 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,065 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,117 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:37,117 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:49,230 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:49,231 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:49,268 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:49,269 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,830 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,831 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,832 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,832 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,832 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,833 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,833 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,834 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,834 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,835 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,835 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,836 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,888 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:56:51,889 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:13,061 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:13,061 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:13,760 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:13,761 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:15,614 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:15,614 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:15,878 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:15,879 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:17,129 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:17,131 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:55,427 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:55,427 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:55,474 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:57:55,474 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 14:58:17,080 - emsawd.core.services - INFO - Fetching weather data for cape town, period: 2025-08-27 to 2025-09-03, years: 5
2025-09-03 14:58:17,080 - emsawd.repositories.geocoding_repository - INFO - Requesting coordinates for 'cape town' from https://geocoding-api.open-meteo.com/v1/search
2025-09-03 14:58:18,802 - emsawd.repositories.geocoding_repository - INFO - API response status: 200
2025-09-03 14:58:18,803 - emsawd.repositories.geocoding_repository - INFO - Successfully found coordinates for 'cape town': (-33.92584, 18.42322)
2025-09-03 14:58:18,805 - emsawd.core.services - INFO - Coordinates for cape town: lat=-33.92584, lon=18.42322
2025-09-03 14:58:18,806 - emsawd.core.services - INFO - Fetching data for year offset 0 (2025): 2025-08-27 to 2025-09-03
2025-09-03 14:58:18,806 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2025-08-27 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:20,150 - emsawd.repositories.openweather_repository - ERROR - Network error for 2025-08-27: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:20,152 - emsawd.core.services - ERROR - Failed to fetch data for 2025: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:20,154 - emsawd.core.services - INFO - Fetching data for year offset 1 (2024): 2024-08-27 to 2024-09-03
2025-09-03 14:58:20,154 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2024-08-27 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:21,380 - emsawd.repositories.openweather_repository - ERROR - Network error for 2024-08-27: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:21,381 - emsawd.core.services - ERROR - Failed to fetch data for 2024: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:21,382 - emsawd.core.services - INFO - Fetching data for year offset 2 (2023): 2023-08-28 to 2023-09-04
2025-09-03 14:58:21,382 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2023-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:24,311 - emsawd.repositories.openweather_repository - ERROR - Network error for 2023-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:24,313 - emsawd.core.services - ERROR - Failed to fetch data for 2023: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:24,315 - emsawd.core.services - INFO - Fetching data for year offset 3 (2022): 2022-08-28 to 2022-09-04
2025-09-03 14:58:24,316 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2022-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:25,610 - emsawd.repositories.openweather_repository - ERROR - Network error for 2022-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:25,611 - emsawd.core.services - ERROR - Failed to fetch data for 2022: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:25,612 - emsawd.core.services - INFO - Fetching data for year offset 4 (2021): 2021-08-28 to 2021-09-04
2025-09-03 14:58:25,612 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2021-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:26,861 - emsawd.repositories.openweather_repository - ERROR - Network error for 2021-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:26,862 - emsawd.core.services - ERROR - Failed to fetch data for 2021: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:26,863 - emsawd.core.services - INFO - Fetching data for year offset 5 (2020): 2020-08-28 to 2020-09-04
2025-09-03 14:58:26,863 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2020-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:28,155 - emsawd.repositories.openweather_repository - ERROR - Network error for 2020-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:28,156 - emsawd.core.services - ERROR - Failed to fetch data for 2020: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:28,157 - emsawd.core.services - INFO - Total records collected: 0
2025-09-03 14:58:37,649 - emsawd.core.services - INFO - Fetching weather data for cape town, period: 2025-08-27 to 2025-09-03, years: 5
2025-09-03 14:58:37,650 - emsawd.repositories.geocoding_repository - INFO - Requesting coordinates for 'cape town' from https://geocoding-api.open-meteo.com/v1/search
2025-09-03 14:58:38,405 - emsawd.repositories.geocoding_repository - INFO - API response status: 200
2025-09-03 14:58:38,405 - emsawd.repositories.geocoding_repository - INFO - Successfully found coordinates for 'cape town': (-33.92584, 18.42322)
2025-09-03 14:58:38,406 - emsawd.core.services - INFO - Coordinates for cape town: lat=-33.92584, lon=18.42322
2025-09-03 14:58:38,407 - emsawd.core.services - INFO - Fetching data for year offset 0 (2025): 2025-08-27 to 2025-09-03
2025-09-03 14:58:38,407 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2025-08-27 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:39,607 - emsawd.repositories.openweather_repository - ERROR - Network error for 2025-08-27: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:39,609 - emsawd.core.services - ERROR - Failed to fetch data for 2025: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:39,611 - emsawd.core.services - INFO - Fetching data for year offset 1 (2024): 2024-08-27 to 2024-09-03
2025-09-03 14:58:39,611 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2024-08-27 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:40,858 - emsawd.repositories.openweather_repository - ERROR - Network error for 2024-08-27: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:40,860 - emsawd.core.services - ERROR - Failed to fetch data for 2024: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:40,862 - emsawd.core.services - INFO - Fetching data for year offset 2 (2023): 2023-08-28 to 2023-09-04
2025-09-03 14:58:40,862 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2023-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:42,118 - emsawd.repositories.openweather_repository - ERROR - Network error for 2023-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:42,119 - emsawd.core.services - ERROR - Failed to fetch data for 2023: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:42,120 - emsawd.core.services - INFO - Fetching data for year offset 3 (2022): 2022-08-28 to 2022-09-04
2025-09-03 14:58:42,121 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2022-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:44,370 - emsawd.repositories.openweather_repository - ERROR - Network error for 2022-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:44,370 - emsawd.core.services - ERROR - Failed to fetch data for 2022: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:44,372 - emsawd.core.services - INFO - Fetching data for year offset 4 (2021): 2021-08-28 to 2021-09-04
2025-09-03 14:58:44,372 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2021-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:45,622 - emsawd.repositories.openweather_repository - ERROR - Network error for 2021-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:45,622 - emsawd.core.services - ERROR - Failed to fetch data for 2021: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:45,625 - emsawd.core.services - INFO - Fetching data for year offset 5 (2020): 2020-08-28 to 2020-09-04
2025-09-03 14:58:45,625 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2020-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 14:58:46,866 - emsawd.repositories.openweather_repository - ERROR - Network error for 2020-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:46,868 - emsawd.core.services - ERROR - Failed to fetch data for 2020: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 14:58:46,870 - emsawd.core.services - INFO - Total records collected: 0
2025-09-03 15:02:34,361 - emsawd.core.services - INFO - Fetching weather data for cape town, period: 2025-08-27 to 2025-09-03, years: 5
2025-09-03 15:02:34,362 - emsawd.repositories.geocoding_repository - INFO - Requesting coordinates for 'cape town' from https://geocoding-api.open-meteo.com/v1/search
2025-09-03 15:02:35,142 - emsawd.repositories.geocoding_repository - INFO - API response status: 200
2025-09-03 15:02:35,142 - emsawd.repositories.geocoding_repository - INFO - Successfully found coordinates for 'cape town': (-33.92584, 18.42322)
2025-09-03 15:02:35,143 - emsawd.core.services - INFO - Coordinates for cape town: lat=-33.92584, lon=18.42322
2025-09-03 15:02:35,143 - emsawd.core.services - INFO - Fetching data for year offset 0 (2025): 2025-08-27 to 2025-09-03
2025-09-03 15:02:35,143 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2025-08-27 at -33.92584,18.42322
2025-09-03 15:02:36,680 - emsawd.repositories.weatherapi_repository - INFO - API response status: 200
2025-09-03 15:02:36,681 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2025-08-28 at -33.92584,18.42322
2025-09-03 15:02:37,508 - emsawd.repositories.weatherapi_repository - INFO - API response status: 200
2025-09-03 15:02:37,510 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2025-08-29 at -33.92584,18.42322
2025-09-03 15:02:39,436 - emsawd.repositories.weatherapi_repository - INFO - API response status: 200
2025-09-03 15:02:39,438 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2025-08-30 at -33.92584,18.42322
2025-09-03 15:02:40,267 - emsawd.repositories.weatherapi_repository - INFO - API response status: 200
2025-09-03 15:02:40,267 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2025-08-31 at -33.92584,18.42322
2025-09-03 15:02:41,116 - emsawd.repositories.weatherapi_repository - INFO - API response status: 200
2025-09-03 15:02:41,116 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2025-09-01 at -33.92584,18.42322
2025-09-03 15:02:41,959 - emsawd.repositories.weatherapi_repository - INFO - API response status: 200
2025-09-03 15:02:41,960 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2025-09-02 at -33.92584,18.42322
2025-09-03 15:02:42,795 - emsawd.repositories.weatherapi_repository - INFO - API response status: 200
2025-09-03 15:02:42,796 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2025-09-03 at -33.92584,18.42322
2025-09-03 15:02:43,673 - emsawd.repositories.weatherapi_repository - INFO - API response status: 200
2025-09-03 15:02:43,674 - emsawd.repositories.weatherapi_repository - INFO - Successfully parsed 8 records from WeatherAPI.
2025-09-03 15:02:43,675 - emsawd.core.services - INFO - Fetched 8 records for 2025
2025-09-03 15:02:43,675 - emsawd.core.services - INFO - Fetching data for year offset 1 (2024): 2024-08-27 to 2024-09-03
2025-09-03 15:02:43,675 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2024-08-27 at -33.92584,18.42322
2025-09-03 15:02:44,525 - emsawd.repositories.weatherapi_repository - ERROR - Network error for 2024-08-27: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2024-08-27
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\weatherapi_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2024-08-27
2025-09-03 15:02:44,527 - emsawd.core.services - ERROR - Failed to fetch data for 2024: Network error while fetching weather data: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2024-08-27
2025-09-03 15:02:44,529 - emsawd.core.services - INFO - Fetching data for year offset 2 (2023): 2023-08-28 to 2023-09-04
2025-09-03 15:02:44,530 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2023-08-28 at -33.92584,18.42322
2025-09-03 15:02:45,350 - emsawd.repositories.weatherapi_repository - ERROR - Network error for 2023-08-28: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2023-08-28
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\weatherapi_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2023-08-28
2025-09-03 15:02:45,351 - emsawd.core.services - ERROR - Failed to fetch data for 2023: Network error while fetching weather data: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2023-08-28
2025-09-03 15:02:45,351 - emsawd.core.services - INFO - Fetching data for year offset 3 (2022): 2022-08-28 to 2022-09-04
2025-09-03 15:02:45,352 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2022-08-28 at -33.92584,18.42322
2025-09-03 15:02:46,221 - emsawd.repositories.weatherapi_repository - ERROR - Network error for 2022-08-28: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2022-08-28
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\weatherapi_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2022-08-28
2025-09-03 15:02:46,222 - emsawd.core.services - ERROR - Failed to fetch data for 2022: Network error while fetching weather data: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2022-08-28
2025-09-03 15:02:46,223 - emsawd.core.services - INFO - Fetching data for year offset 4 (2021): 2021-08-28 to 2021-09-04
2025-09-03 15:02:46,223 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2021-08-28 at -33.92584,18.42322
2025-09-03 15:02:47,069 - emsawd.repositories.weatherapi_repository - ERROR - Network error for 2021-08-28: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2021-08-28
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\weatherapi_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2021-08-28
2025-09-03 15:02:47,070 - emsawd.core.services - ERROR - Failed to fetch data for 2021: Network error while fetching weather data: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2021-08-28
2025-09-03 15:02:47,071 - emsawd.core.services - INFO - Fetching data for year offset 5 (2020): 2020-08-28 to 2020-09-04
2025-09-03 15:02:47,071 - emsawd.repositories.weatherapi_repository - INFO - Requesting WeatherAPI for 2020-08-28 at -33.92584,18.42322
2025-09-03 15:02:47,895 - emsawd.repositories.weatherapi_repository - ERROR - Network error for 2020-08-28: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2020-08-28
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\weatherapi_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2020-08-28
2025-09-03 15:02:47,897 - emsawd.core.services - ERROR - Failed to fetch data for 2020: Network error while fetching weather data: 400 Client Error: Bad Request for url: https://api.weatherapi.com/v1/history.json?key=6b56cbe53d7c4924aeb130157250309&q=-33.92584%2C18.42322&dt=2020-08-28
2025-09-03 15:02:47,900 - emsawd.core.services - INFO - Total records collected: 8
2025-09-03 15:02:47,914 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:02:47,914 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:02:47,962 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:02:47,963 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:03:01,033 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:03:01,034 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:03:01,098 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:03:01,098 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:03:15,645 - emsawd.core.services - INFO - Fetching weather data for cape town, period: 2025-08-27 to 2025-09-03, years: 5
2025-09-03 15:03:15,646 - emsawd.repositories.geocoding_repository - INFO - Requesting coordinates for 'cape town' from https://geocoding-api.open-meteo.com/v1/search
2025-09-03 15:03:16,413 - emsawd.repositories.geocoding_repository - INFO - API response status: 200
2025-09-03 15:03:16,413 - emsawd.repositories.geocoding_repository - INFO - Successfully found coordinates for 'cape town': (-33.92584, 18.42322)
2025-09-03 15:03:16,415 - emsawd.core.services - INFO - Coordinates for cape town: lat=-33.92584, lon=18.42322
2025-09-03 15:03:16,415 - emsawd.core.services - INFO - Fetching data for year offset 0 (2025): 2025-08-27 to 2025-09-03
2025-09-03 15:03:16,415 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2025-08-27 at lat=-33.92584, lon=18.42322
2025-09-03 15:03:17,692 - emsawd.repositories.openweather_repository - ERROR - Network error for 2025-08-27: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:17,694 - emsawd.core.services - ERROR - Failed to fetch data for 2025: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1756288800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:17,696 - emsawd.core.services - INFO - Fetching data for year offset 1 (2024): 2024-08-27 to 2024-09-03
2025-09-03 15:03:17,696 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2024-08-27 at lat=-33.92584, lon=18.42322
2025-09-03 15:03:18,994 - emsawd.repositories.openweather_repository - ERROR - Network error for 2024-08-27: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:18,996 - emsawd.core.services - ERROR - Failed to fetch data for 2024: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1724752800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:18,998 - emsawd.core.services - INFO - Fetching data for year offset 2 (2023): 2023-08-28 to 2023-09-04
2025-09-03 15:03:18,999 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2023-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 15:03:20,241 - emsawd.repositories.openweather_repository - ERROR - Network error for 2023-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:20,242 - emsawd.core.services - ERROR - Failed to fetch data for 2023: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1693216800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:20,244 - emsawd.core.services - INFO - Fetching data for year offset 3 (2022): 2022-08-28 to 2022-09-04
2025-09-03 15:03:20,244 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2022-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 15:03:21,546 - emsawd.repositories.openweather_repository - ERROR - Network error for 2022-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:21,547 - emsawd.core.services - ERROR - Failed to fetch data for 2022: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1661680800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:21,548 - emsawd.core.services - INFO - Fetching data for year offset 4 (2021): 2021-08-28 to 2021-09-04
2025-09-03 15:03:21,548 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2021-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 15:03:24,576 - emsawd.repositories.openweather_repository - ERROR - Network error for 2021-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:24,578 - emsawd.core.services - ERROR - Failed to fetch data for 2021: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1630144800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:24,581 - emsawd.core.services - INFO - Fetching data for year offset 5 (2020): 2020-08-28 to 2020-09-04
2025-09-03 15:03:24,581 - emsawd.repositories.openweather_repository - INFO - Requesting OpenWeather for 2020-08-28 at lat=-33.92584, lon=18.42322
2025-09-03 15:03:25,764 - emsawd.repositories.openweather_repository - ERROR - Network error for 2020-08-28: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\emsawd\repositories\openweather_repository.py", line 54, in get_historical_weather
    response.raise_for_status()
  File "C:\Users\User\OneDrive\Documents\GitHub\enhanced-multi-service-aggregated-weather-data\.venv\Lib\site-packages\requests\models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:25,765 - emsawd.core.services - ERROR - Failed to fetch data for 2020: Network error while fetching weather data: 401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall/timemachine?lat=-33.92584&lon=18.42322&dt=1598608800&appid=d6a67f5f0d27c66d0fa74b597218856b
2025-09-03 15:03:25,766 - emsawd.core.services - INFO - Total records collected: 0
2025-09-03 15:04:25,834 - emsawd.core.services - INFO - Fetching weather data for cape town, period: 2025-08-20 to 2025-09-03, years: 5
2025-09-03 15:04:25,834 - emsawd.repositories.geocoding_repository - INFO - Requesting coordinates for 'cape town' from https://geocoding-api.open-meteo.com/v1/search
2025-09-03 15:04:26,632 - emsawd.repositories.geocoding_repository - INFO - API response status: 200
2025-09-03 15:04:26,632 - emsawd.repositories.geocoding_repository - INFO - Successfully found coordinates for 'cape town': (-33.92584, 18.42322)
2025-09-03 15:04:26,633 - emsawd.core.services - INFO - Coordinates for cape town: lat=-33.92584, lon=18.42322
2025-09-03 15:04:26,633 - emsawd.core.services - INFO - Fetching data for year offset 0 (2025): 2025-08-20 to 2025-09-03
2025-09-03 15:04:26,633 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 15:04:27,452 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 15:04:27,453 - emsawd.repositories.weather_repository - INFO - Successfully parsed 15 records from API response.
2025-09-03 15:04:27,454 - emsawd.core.services - INFO - Fetched 15 records for 2025
2025-09-03 15:04:27,454 - emsawd.core.services - INFO - Fetching data for year offset 1 (2024): 2024-08-20 to 2024-09-03
2025-09-03 15:04:27,454 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 15:04:28,206 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 15:04:28,207 - emsawd.repositories.weather_repository - INFO - Successfully parsed 15 records from API response.
2025-09-03 15:04:28,207 - emsawd.core.services - INFO - Fetched 15 records for 2024
2025-09-03 15:04:28,207 - emsawd.core.services - INFO - Fetching data for year offset 2 (2023): 2023-08-21 to 2023-09-04
2025-09-03 15:04:28,209 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 15:04:28,986 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 15:04:28,987 - emsawd.repositories.weather_repository - INFO - Successfully parsed 15 records from API response.
2025-09-03 15:04:28,988 - emsawd.core.services - INFO - Fetched 15 records for 2023
2025-09-03 15:04:28,988 - emsawd.core.services - INFO - Fetching data for year offset 3 (2022): 2022-08-21 to 2022-09-04
2025-09-03 15:04:28,988 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 15:04:30,454 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 15:04:30,455 - emsawd.repositories.weather_repository - INFO - Successfully parsed 15 records from API response.
2025-09-03 15:04:30,456 - emsawd.core.services - INFO - Fetched 15 records for 2022
2025-09-03 15:04:30,456 - emsawd.core.services - INFO - Fetching data for year offset 4 (2021): 2021-08-21 to 2021-09-04
2025-09-03 15:04:30,456 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 15:04:31,211 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 15:04:31,212 - emsawd.repositories.weather_repository - INFO - Successfully parsed 15 records from API response.
2025-09-03 15:04:31,214 - emsawd.core.services - INFO - Fetched 15 records for 2021
2025-09-03 15:04:31,215 - emsawd.core.services - INFO - Fetching data for year offset 5 (2020): 2020-08-21 to 2020-09-04
2025-09-03 15:04:31,215 - emsawd.repositories.weather_repository - INFO - Requesting historical weather for lat=-33.92584, lon=18.42322
2025-09-03 15:04:31,982 - emsawd.repositories.weather_repository - INFO - API response status: 200
2025-09-03 15:04:31,982 - emsawd.repositories.weather_repository - INFO - Successfully parsed 15 records from API response.
2025-09-03 15:04:31,984 - emsawd.core.services - INFO - Fetched 15 records for 2020
2025-09-03 15:04:31,984 - emsawd.core.services - INFO - Total records collected: 90
2025-09-03 15:04:32,001 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,002 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,003 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,004 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,004 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,005 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,005 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,006 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,007 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,008 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,009 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,009 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,074 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:04:32,076 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:06,039 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:06,041 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:07,548 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:07,548 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:08,558 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:08,559 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:09,172 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:09,173 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:09,720 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:09,721 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:12,167 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:12,168 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:12,526 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:12,527 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:12,846 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:12,847 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:13,261 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:13,262 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:13,735 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:13,737 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:20,048 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:20,048 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:20,102 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:05:20,102 - matplotlib.category - INFO - Using categorical units to plot a list of strings that are all parsable as floats or dates. If these strings should be plotted as numbers, cast to the appropriate data type before plotting.
2025-09-03 15:06:50,932 - root - INFO - Application exiting with code 0.
